featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -94,3 +105,82 @@ def serialize_name_unchanged(original):
 
     for feature_type in feature_type_list:
         serialize_name_unchanged(feature_type)
+
+
+def test_deserialize_features_s3(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    features_deserialized = sorted(ft.load_features(S3_URL), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_anon_features_s3(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    features_deserialized = sorted(ft.load_features(S3_URL, profile_name=False), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_deserialize_features_url(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    features_deserialized = sorted(ft.load_features(URL), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_serialize_url(es):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+    error_text = "Writing to URLs is not supported"
+    with pytest.raises(ValueError, match=error_text):
+        ft.save_features(features_original, URL)
+
+
+def tests_s3_profile_serialize(es):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+    error_text = "The config profile (.*) could not be found"
+    with pytest.raises(ProfileNotFound, match=error_text):
+        ft.save_features(features_original, S3_URL, profile_name=TEST_CONFIG)
+
+
+def tests_s3_profile_deserialize(es):
+    error_text = "The config profile (.*) could not be found"
+    with pytest.raises(ProfileNotFound, match=error_text):
+        ft.load_features(S3_URL, profile_name=TEST_CONFIG)
+
+
+@pytest.fixture
+def s3_client():
+    from moto import mock_s3
```

Comment:
I had to run the import in here as importing mock_s3 would set environment variables to moto test access and secret access keys.